flink/v1.14/flink/src/main/java/org/apache/iceberg/flink/data/FlinkParquetReaders.java

```diff
@@ -284,7 +284,6 @@ private FlinkParquetReaders() {
     public DecimalData read(DecimalData ignored) {
       Binary binary = column.nextBinary();
       BigDecimal bigDecimal = new BigDecimal(new BigInteger(binary.getBytes()), scale);
-      // TODO: need a unit test to write-read-validate decimal via FlinkParquetWrite/Reader
```

Comment:
If this is what is being addressed by this PR, can you add decimal to the description and summary?

Reply:
The org.apache.iceberg.data.DataTest#testSimpleStruct already contains data of decimal type, so I did not add a separate test case for that.
https://github.com/apache/iceberg/blob/1474f8408aefc031ec271399859bc51cb2e96f8e/data/src/test/java/org/apache/iceberg/data/DataTest.java#L42-L68

Reply:
This PR is more about using FlinkParquetWrite and Reader to write and read data at the same time, rather than using either of them alone.